src/basic/terminal-util.c

```diff
@@ -455,7 +455,7 @@ int acquire_terminal(
                                         goto fail;
                                 }
 
-                                r = fd_wait_for_event(fd, POLLIN, ts + timeout - n);
+                                r = fd_wait_for_event(notify, POLLIN, ts + timeout - n);
```

Comment:
@poettering sorry I wasn't clear enough about the lockfile.

I was thinking about lock files which would be similar to the one created by lockfile(1): a regular file which is created by the first process asking the confirmation. Other concurrent processes would detect this lock file, monitor it and go to sleep until this file is removed.

This way we could precisely release the lock and wouldn't be subject to any close() on the tty which can happen at nearly any time.

But that would also require us to create a special  _acquire_terminal()_ function mostly dedicated to the confirm_spawn=yes code.